presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -231,6 +233,45 @@ public void testHiveViewInInformationSchema()
                 .contains(row("n_nationkey", "bigint", "", ""));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testFailingHiveViewsForInformationSchema()
+    {
+        if (getHiveVersionMajor() < 3) {
+            // Skip older versions since they don't support "getTableNamesByType" in ThriftHiveMetastore#getAllViews
+            throw new SkipException("Only test the behavior for hive 3");
+        }
+
+        onHive().executeQuery("DROP SCHEMA IF EXISTS test_list_failing_views CASCADE");
+        onHive().executeQuery("CREATE SCHEMA test_list_failing_views");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.correct_view AS SELECT * FROM nation limit 5");
+
+        // Create a view for which the translation is guaranteed to fail
+        onPresto().executeQuery("CREATE TABLE test_list_failing_views.table_dropped (col0 BIGINT)");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.failing_view AS SELECT * FROM test_list_failing_views.table_dropped");
+        onPresto().executeQuery("DROP TABLE test_list_failing_views.table_dropped");
+
+        assertThat(query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views'"))
+                .containsOnly(row("correct_view"));
+
+        assertThat(query("SELECT table_name FROM information_schema.views"))
+                .contains(row("correct_view"));
+
+        assertThat(query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views' and table_name = 'correct_view'"))
+                .containsOnly(row("correct_view"));
+
+        // Listing fails when metadata for the problematic view is queried specifically
+        assertThatThrownBy(() -> query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views' and table_name = 'failing_view'"))
+                .hasMessageContaining("Failed to translate Hive view 'test_list_failing_views.failing_view'");
+
+        // Queries on information_schema.columns also trigger ConnectorMetadata#getViews. Columns from failing_view are
+        // listed too since HiveMetadata#listTableColumns does not ignore views.
+        assertThat(query("SELECT DISTINCT table_name FROM information_schema.columns WHERE table_schema = 'test_list_failing_views'"))
+                .containsOnly(row("correct_view"), row("failing_view"));
```

Comment:
`DISTINCT table_name` -> table_name, column_name

so that we test also _what_ is returned for these views